flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/PushPartitionIntoTableSourceScanRule.scala

```diff
@@ -112,14 +124,16 @@ class PushPartitionIntoTableSourceScanRule extends RelOptRule(
     }
 
     val statistic = tableSourceTable.getStatistic
-    val newStatistic = if (remainingPartitions.size() == allPartitions.size()) {
-      // Keep all Statistics if no predicates can be pushed down
-      statistic
-    } else if (statistic == FlinkStatistic.UNKNOWN) {
-      statistic
-    } else {
-      // Remove tableStats after predicates pushed down
-      FlinkStatistic.builder().statistic(statistic).tableStats(null).build()
+    val newStatistic = {
+      val tableStats = catalogOption match {
```

Comment:
if no partitions are pruned and the original `TableStats` is not `UNKNOWN`, we could use the original `TableStats` and avoid fetch all partitions' statistics

Reply:
- First get stats from regular table stats may be not correct.
- In the next step, I want to introduce `listPartitionsByFilter`, so we can not know the msg: `if no partitions are pruned`.